src/js/events/MetronomeActions.js

```diff
@@ -74,10 +65,21 @@ const MetronomeActions = {
               });
               resolve();
             } catch (error) {
-              handleError(error);
+              AppDispatcher.handleServerAction({
+                type: REQUEST_METRONOME_JOBS_ERROR,
+                data: error
+              });
+              reject();
             }
           },
-          error: handleError,
+          error(xhr) {
+            AppDispatcher.handleServerAction({
+              type: REQUEST_METRONOME_JOBS_ERROR,
+              data: xhr.message,
```

Comment:
Please adjust the `REQUEST_METRONOME_JOBS_ERROR` event as well as similar cases to always provide the same type of data as varying types result in unnecessary complex handling logic. It might be worth to think about a general error event object (`{code, message}`)  to provide a consistent interface for error handling.